packages/runtime/src/registry.tsx

```diff
@@ -49,18 +50,25 @@ export type ComponentImplementation<T = any> = React.FC<
     mergeState: MergeState;
     subscribeMethods: SubscribeMethods;
     slotsMap: SlotsMap | undefined;
+    style?: CSSProperties;
+    data?: Record<string, any>;
   }
 >;
 
+export type TraitResult = {
+  props: {
+    data?: unknown;
```

Comment:
Looks like we need to improve the spec so the trait can declare what kind of props it will return to the component?

Reply:
Yes we can.And the component need declare what kind of props it can deal with. But I think the trait result interface may change. We can do it after the interface become stable.